src/js_mixed.hpp

```diff
@@ -0,0 +1,187 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <common/type_deduction.hpp>
+#include <iostream>
+#include <map>
+#include <realm/mixed.hpp>
+#include <type_traits>
+
+namespace realm {
+namespace js {
+
+template <typename Context, typename Value>
+class MixedWrapper {
+   public:
+    virtual Mixed wrap(Context, Value const &) = 0;
+    virtual Value unwrap(Context, Mixed) = 0;
+};
+
+template <typename Context, typename Value, typename Utils>
+class HandleString : public MixedWrapper<Context, Value> {
```

Comment:
I think these classes would be better named Mixed* (e.g., MixedString) rather than Handle*.  Just to make it clear that they deal with Mixed values only.